vuestorefront/schemas/product.py

```diff
@@ -130,23 +130,22 @@ class ProductQuery(graphene.ObjectType):
     @staticmethod
     def resolve_product(self, info, id=None, slug=None, barcode=None):
         env = info.context["env"]
-        Product = env["product.template"].sudo()
-
+        Product = product = env["product.template"].sudo()
+        domain = None
+        base_domain = [('is_published', '=', True)]
```

Comment:
Would it be possible to access a product that is not published in special cases when such a thing is needed? 
I couldn't see such a change in the project-specific module

Reply:
So we should have backdoor for unpublished products?:) I guess I can just add a flag/filter, `only_published=True`. So you can pass `only_published=False` and it would not limit to published only?

Reply:
IMO Public queries should have access to only the data that is necessary for the client system to function. 
In most of the projects would be needed only published products, as it is now. Just we need to make it easily customizable for projects where there are different cases and do the update there.

Reply:
Another way to check if a user can see a certain single product is by adding `_access_check` which can include additional checks for custom projects.